### PR TITLE
Integrate new authentication

### PR DESCRIPTION
**Pull request by ptbrowne**
The updated cozy-authentication library takes care of most of the work done initially in the MobileRouter. This and the fact that cozy-client has learned to emit events let us organize the code in a more logical and decoupled way.

For example, it is now the responsibility of cozy-authentication to save the credentials and reconnect the user if the app is reopened after a first login. This enables us to completely remove the mobile store that was used to store user credentials.

Thanks to events, it is possible to decouple the code used by the bar to init from the initialisation of the client. Same for the push notification.

### CODE REVIEW

File: src/ducks/mobile/bar.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    if (document.getElementById('coz-bar')) {
+      document.getElementById('coz-bar').remove()
+      document.body.setAttribute('style', '')
+    }
+  })
+
+  client.on('refreshToken', () => {
+    cozy.bar.updateAccessToken(client.stackClient.token.accessToken)
+  })
+}
```
Comment (drazik) [inline src/ducks/mobile/bar.js]: :+1:

File: src/ducks/bar/BarTheme.jsx [hunk 2]
```diff
@@ -1,7 +1,21 @@
+/* global cozy, __TARGET__ */
+
 import React from 'react'
 import PropTypes from 'prop-types'
 import withSideEffect from 'react-side-effect'
-import { setBarTheme } from 'ducks/mobile/utils'
+import { setTheme as setStatusBarTheme } from './statusBar'
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply src/ducks/bar/BarTheme.jsx]: Done thanks !

File: src/ducks/mobile/bar.js (hunk 1 above)
Comment (ptbrowne) [reply src/ducks/mobile/bar.js]: Not needed anymore since the bar is initaliazed with a CozyClient, we can completely remove this !

Comment (ptbrowne) [reply src/ducks/mobile/bar.js]: Done in https://github.com/cozy/cozy-banks/pull/1177/commits/88b5895279ac215d97eb1dccd32e41d5dac7ecc5

File: package.json [hunk 3]
```diff
@@ -174,6 +174,7 @@
     "geco": "0.11.1",
     "hammerjs": "2.0.8",
     "html-webpack-harddisk-plugin": "1.0.1",
+    "ios-sim": "8.0.1",
```
Comment (kosssi) [inline package.json]: why you import this dep?

File: src/ducks/pin/PinAuth.jsx [hunk 4]
```diff
@@ -119,9 +118,8 @@ class PinAuth extends React.Component {
   }
 
   logout() {
-    const { router } = this.props
-    const { client, store } = this.context
-    onLogout(store, client, router)
+    const client = this.props.client
```
Comment (kosssi) [inline src/ducks/pin/PinAuth.jsx]:
```suggestion
    const { client } = this.props
```

File: src/ducks/client/mobile.js [hunk 5]
```diff
@@ -48,6 +54,7 @@ export const getManifestOptions = manifest => {
   }
 }
 
+// Should be moved to cozy-pouch
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply src/ducks/client/mobile.js]: OK for the discussion, I will update the comment to // Should be moved to a shared repo

File: src/ducks/pin/PinAuth.jsx (hunk 4 above)
Comment (ptbrowne) [reply src/ducks/pin/PinAuth.jsx]: Done

File: package.json (hunk 3 above)
Comment (ptbrowne) [reply package.json]: yarn ios:run a planté si je faisais pas ca.

File: src/targets/mobile/fastlane/Fastfile [hunk 6]
```diff
@@ -4,7 +4,8 @@ platform :ios do
   lane :devices do
     register_devices(
        devices: {
-         "Patrick iPhone 6s" => "dbb0aa403f6928368467d3f10b28a61336e52023"
+         "Patrick iPhone 6s" => "dbb0aa403f6928368467d3f10b28a61336e52023",
+         "Quentin iPhone X" => "f104c368769384e8f043ba497a74f704d27d6d3f"
```
Comment (drazik) [inline src/targets/mobile/fastlane/Fastfile]: What is it useful for? (just asking for my culture :))

Comment (ptbrowne) [reply src/targets/mobile/fastlane/Fastfile]: To reigster devices on itunes connect.

**Pull request merged**